test/MultiOwnable/RemoveOwnerAtIndex.t.sol

```diff
@@ -6,39 +6,47 @@ import "./MultiOwnableTestBase.t.sol";
 contract RemoveOwnerAtIndexTest is MultiOwnableTestBase {
     function testRemovesOwner() public {
         vm.prank(owner1Address);
-        _removeOwner();
+        _removeOwner(owner2Bytes);
         assertFalse(mock.isOwnerBytes(owner2Bytes));
     }
 
     function testRemovesOwnerAtIndex() public {
         uint8 index = _index();
         vm.prank(owner1Address);
-        _removeOwner();
+        _removeOwner(owner2Bytes);
         assertEq(mock.ownerAtIndex(index), hex"");
     }
 
     function testEmitsRemoveOwner() public {
         vm.expectEmit(true, true, true, true);
         emit MultiOwnable.RemoveOwner(_index(), owner2Bytes);
         vm.prank(owner1Address);
-        _removeOwner();
+        _removeOwner(owner2Bytes);
     }
 
-    function testRevertsIfCalledByNonOwner() public {
-        vm.startPrank(address(0xdead));
+    function testRevertsIfCalledByNonOwner(address a) public {
+        vm.assume(a != owner1Address);
+        vm.startPrank(a);
         vm.expectRevert(MultiOwnable.Unauthorized.selector);
-        _removeOwner();
+        _removeOwner(abi.encode(a));
     }
 
     function testRevertsIfNoOwnerAtIndex() public {
         uint8 index = 10;
         vm.expectRevert(abi.encodeWithSelector(MultiOwnable.NoOwnerAtIndex.selector, index));
         vm.prank(owner1Address);
-        mock.removeOwnerAtIndex(index);
+        mock.removeOwnerAtIndex(index, owner1Bytes);
     }
 
-    function _removeOwner() internal virtual {
-        mock.removeOwnerAtIndex(_index());
+    function testRevertsIfWrongOwnerAtIndex() public {
+        uint8 index = _index();
+        vm.expectRevert(abi.encodeWithSelector(MultiOwnable.WrongOwnerAtIndex.selector, index, owner1Bytes));
+        vm.prank(owner1Address);
+        mock.removeOwnerAtIndex(index, owner1Bytes);
+    }
+
+    function _removeOwner(bytes memory owner) internal virtual {
```

Comment:
looks like we're barely using the function? Maybe remove. Think is carryover from the inheritance stuff